### PR TITLE
[simdjson] Update to v0.9.6

### PR DESCRIPTION
**Pull request by myd7349**
**Describe the pull request**

- #### What does your PR fix?  
  Fixes #...

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  <all / linux, windows, ...>, <Yes/No>

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  `Your answer`

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  <Yes / I am still working on this PR>

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

### CODE REVIEW

File: ports/simdjson/portfile.cmake [hunk 1]
```diff
@@ -1,23 +1,28 @@
...
-        -DSIMDJSON_BUILD_STATIC=${SIMDJSON_BUILD_STATIC}
-        -DSIMDJSON_IMPLEMENTATION_ARM64=${SIMDJSON_IMPLEMENTATION_ARM64}
-        -DSIMDJSON_JUST_LIBRARY=ON
+        -DSIMDJSON_DEVELOPMENT_CHECKS=OFF
+        -DSIMDJSON_DEVELOPER_MODE=OFF
         -DSIMDJSON_GOOGLE_BENCHMARKS=OFF
         -DSIMDJSON_COMPETITION=OFF
-        -DSIMDJSON_SANITIZE=OFF # issue 10145, pr 11495
+        -DSIMDJSON_ENABLE_FUZZING=OFF
+        -DSIMDJSON_CXXOPTS=OFF
```
Comment (JonLiu1993) [inline ports/simdjson/portfile.cmake:24]: @myd7349 ,Thanks for posting this pr,These options don't seem to be found in CMakeLists

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

**Comment by JonLiu1993** [comment]
@myd7349 ,Please let me know if this pr is ready for review

**Comment by myd7349** [comment]
Hi! @JonLiu1993 Thanks for your helpful review and sorry for the long delay.
This PR is ready now.

File: ports/simdjson/portfile.cmake [hunk 2]
```diff
@@ -1,23 +1,32 @@
...
+    REF e9b893ff1b13c6a70135827c62b3f3d65938d135 # v0.9.6
     HEAD_REF master
-    SHA512 2c033e8fa253103f00606b7a14502d19c06385f22d1d09ea4b1edc6f443ba51ee95d49c790a05a64e28cd03f4350118b23c3c7f759a60e26a43da658609a5317
+    SHA512 977b92ffae7219680f3d8567b1911b0d17ac1143a2ba58d7a4007cdcbf42dca6362853fcf3c3caf4af2029bc5f6a3cb8fab6139050a9d8539e8e4c7df646837d
+)
+
+vcpkg_check_features(
+    OUT_FEATURE_OPTIONS FEATURE_OPTIONS
+    FEATURES
+        exceptions    SIMDJSON_EXCEPTIONS
```
Comment (JonLiu1993) [inline ports/simdjson/portfile.cmake:12]: @myd7349 ,SIMDJSON_EXCEPTIONS is not found in CMakeLists, where is this feature implemented

**Comment by JonLiu1993** [comment]
@myd7349, have you tested these functions locally?

**Comment by myd7349** [comment]
> 
> 
> @myd7349, have you tested these functions locally?

Hi! @JonLiu1993 I have tested these features on:
- x86-windows
- x64-windows
- x64-windows-static

and it worked as expected on all these platforms.

My linux machine doesn't have a C++17 compiler, so I was not able to test them on `x64-linux`.

**Review by JonLiu1993 — APPROVED** [review]

File: ports/simdjson/portfile.cmake [hunk 3]
```diff
@@ -1,23 +1,32 @@
...
+        no-deprecated SIMDJSON_DISABLE_DEPRECATED_API
```
Comment (BillyONeal) [inline ports/simdjson/portfile.cmake:13]: Features need to be additive, not subtractive; e.g. port a that depends on simdjson[no-deprecated] and port b that depends on simdjson together produce non-functional results.

**Review by BillyONeal — CHANGES_REQUESTED** [review]

File: ports/simdjson/portfile.cmake (hunk 3 above)
Comment (BillyONeal) [reply ports/simdjson/portfile.cmake:13]: (If you invert the feature it's ok)

Comment (myd7349) [reply ports/simdjson/portfile.cmake:13]: Hi! @BillyONeal Thanks for your helpful review. That sounds resonable.

**Review by JonLiu1993 — APPROVED** [review]

**Review by BillyONeal — APPROVED** [review]

**Comment by JonLiu1993** [comment]
Fix #19009

**Comment by BillyONeal** [comment]
Thanks for your contribution @myd7349 !

**Pull request merged**